public/app/features/dashboard/containers/PublicDashboardPage.tsx

```diff
@@ -1,13 +1,93 @@
 
-import DashboardPage, { DashboardPageRouteParams, DashboardPageRouteSearchParams } from './DashboardPage';
+import { DashNavTimeControls } from '../components/DashNav/DashNavTimeControls';
+import { DashboardFailed } from '../components/DashboardLoading/DashboardFailed';
+import { DashboardLoading } from '../components/DashboardLoading/DashboardLoading';
+import { PublicDashboardFooter } from '../components/PublicDashboardFooter/PublicDashboardsFooter';
+import { DashboardGrid } from '../dashgrid/DashboardGrid';
+import { getTimeSrv } from '../services/TimeSrv';
+import { DashboardModel } from '../state';
+import { initDashboard } from '../state/initDashboard';
+
+import { DashboardPageRouteParams, DashboardPageRouteSearchParams } from './DashboardPage';
 
 export type Props = GrafanaRouteComponentProps<DashboardPageRouteParams, DashboardPageRouteSearchParams>;
 
+const Toolbar = ({ dashboard }: { dashboard: DashboardModel }) => {
+  const dispatch = useDispatch();
+
+  const onChangeTimeZone = (timeZone: TimeZone) => {
+    dispatch(updateTimeZoneForSession(timeZone));
+  };
+
+  return (
+    <PageToolbar title={dashboard.title}>
+      <ToolbarButtonRow alignment="right">
+        {!dashboard.timepicker.hidden && (
+          <DashNavTimeControls dashboard={dashboard} onChangeTimeZone={onChangeTimeZone} />
+        )}
+      </ToolbarButtonRow>
+    </PageToolbar>
+  );
+};
+
 const PublicDashboardPage = (props: Props) => {
-  return <DashboardPage isPublic {...props} />;
+  const { match, route, location } = props;
+  const dispatch = useDispatch();
+  const context = useGrafana();
+  const prevProps = usePrevious(props);
+  const dashboardState = useSelector((store) => store.dashboard);
+  const dashboard = dashboardState.getModel();
+
+  useEffect(() => {
+    dispatch(
+      initDashboard({
+        routeName: route.routeName,
+        fixUrl: false,
+        accessToken: match.params.accessToken,
+        keybindingSrv: context.keybindings,
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (prevProps?.location.search !== location.search) {
+      const prevUrlParams = prevProps?.queryParams;
+      const urlParams = props.queryParams;
+
+      if (urlParams?.from !== prevUrlParams?.from || urlParams?.to !== prevUrlParams?.to) {
+        getTimeSrv().updateTimeRangeFromUrl();
+      }
+
+      if (!prevUrlParams?.refresh && urlParams?.refresh) {
+        getTimeSrv().setAutoRefresh(urlParams.refresh);
+      }
+    }
+  }, [prevProps, location.search, props.queryParams]);
+
+  if (!dashboard) {
+    return <DashboardLoading initPhase={dashboardState.initPhase} />;
+  }
+
+  return (
+    <>
+      <Page layout={PageLayoutType.Canvas} toolbar={<Toolbar dashboard={dashboard} />}>
+        {dashboardState.initError && <DashboardFailed />}
+        <DashboardGrid dashboard={dashboard} isEditable={false} viewPanel={null} editPanel={null} />
+      </Page>
```

Comment:
This is not working with topnav enabled, the new Page component (when topnav is enabled) is not designed to work with toolbar (as that is part of the AppChrome). 

That was why had custom styles here.

Reply:
the PageToolbar component will be deprecated as well after topnav is launched

Reply:
@torkelo So, you say not to use Page component and make our own styles? Why is not working? I just tested it and it works with the toolbar I pass as prop.

Reply:
@torkelo instead of PageToolbar component, which will be deprecated, what should we use?

Reply:
@juanicabanas does not work with topnav feature enabled